source/compute/mus_compute_d3q19_module.fpp

```diff
@@ -4346,6 +4356,1171 @@ end subroutine f_f_eq_regularized_4th_ord_d3q19
+    real(kind=rk) :: rho, feq
+    real(kind=rk) :: d_omega, nu_q
+    real(kind=rk) :: transVel( nSolve*3 ) ! velocity from the transport field
+    real(kind=rk) :: uc, cqx2, cqy2, cqz2, cq2
+    real(kind=rk) :: a_e, a_xx, a_ww, a_xy, a_xz, a_yz
+    real(kind=rk) :: P_e, P_xx, P_ww, P_xy, P_xz, P_yz
+    integer :: vel_varPos ! position of transport velocity variable in varSys
+    real(kind=rk) :: inv_vel, u_fluid(3)
+    ! --------------------------------------------------------------------------
+    ! access scheme via 1st variable method data which is a state variable
+    call C_F_POINTER( varSys%method%val(derVarPos(1)%pdf)%method_Data, fPtr )
+    scheme => fPtr%solverData%scheme
+
+    ! passive scalar has only one transport Variable
+    vel_varPos = scheme%transVar%method(1)%data_varPos
+    ! Get velocity field
+    call varSys%method%val(vel_varPos)%get_valOfIndex( &
+      & varSys  = varSys,                              &
+      & time    = params%general%simControl%now,       &
+      & iLevel  = level,                               &
+      & idx     = scheme%transVar%method(1)            &
+      &           %pntIndex%indexLvl(level)            &
+      &           %val(1:nSolve),                      &
+      & nVals   = nSolve,                              &
+      & res     = transVel                             )
+
+    ! convert physical velocity into LB velocity
+    inv_vel = 1.0_rk / params%physics%fac( level )%vel
+    transVel = transVel * inv_vel
+
+    ! for isotropic diffusion factor, it turns out to be 1st order bgk
+    d_omega = fieldProp(1)%species%omega
+    ! reciprocal of the free parameter nu, i.e. nu_q = 1/nu
+    nu_q = cs2inv / (1.0_rk / d_omega - 0.5_rk)
+    a_e = ( fieldProp(1)%species%diff_tensor(Dxx)  &
+      &    + fieldProp(1)%species%diff_tensor(Dyy) &
+      &    + fieldProp(1)%species%diff_tensor(Dzz) ) * nu_q / 3.0_rk - 1.0_rk
+    a_xx = 2._rk / 3._rk * nu_q * ( fieldProp(1)%species%diff_tensor(Dxx)              &
+      &                          - 0.5_rk * ( fieldProp(1)%species%diff_tensor(Dyy)    &
+      &                                      + fieldProp(1)%species%diff_tensor(Dzz) ) &
+      &                          )
+    a_ww = 0.5_rk * nu_q * ( fieldProp(1)%species%diff_tensor(Dyy)  &
+      &                     - fieldProp(1)%species%diff_tensor(Dzz) )
+
+    ! attention: D_ab has a multiplier of 2 in ADE
+    a_xy = fieldProp(1)%species%diff_tensor(Dxy) * nu_q
+    a_xz = fieldProp(1)%species%diff_tensor(Dxz) * nu_q
+    a_yz = fieldProp(1)%species%diff_tensor(Dyz) * nu_q
+
+    elemloop: do iElem = 1, nSolve
+      ! x-, y- and z-velocity from transport field
+      u_fluid = transVel( (iElem-1)*3+1 : iElem*3 )
+
+      do iDir = 1, layout%fStencil%QQ
+        pdfTmp( iDir ) =                                  &
+          & instate( neigh (( idir-1)* nelems+ ielem)+    &
+          & ( 1-1)* layout%fstencil%qq+ varsys%nscalars*0 )
```

Comment:
If we need this info for clarity, maybe add a comment. But the multiplication with 0 shouldn't be there in the code. The compiler probably optimizes this out, but we should not rely on that. Also please make the spacing around the operators symmetric.